### PR TITLE
Feature Integration

### PR DESCRIPTION
**Pull request by samfalberg**
These commits will integrate the many features we've been working on. This will make it so the user can navigate the site easily without having to manually change the URL. Commits will include adding things like a homepage and login/logout link to every page. When a user logs in or out, the webapp should navigate them back to the page they were previously on.

### CODE REVIEW

**Comment by samfalberg** [comment]
A homepage link will be in the top left corner of every page. Eventually we'll replace this with the Sullivan logo, which will also direct you to the homepage when clicked.

Screenshot 2020-07-01 at 11 26 15 AM

**Comment by samfalberg** [comment]
A login/logout link will be displayed at the top right of every page. If the user is logged in, it will show the logout link, which will simply logout the user but keep them at the page they are currently on. If the user is logged out, it will show the login link, which will redirect the user to the registration page.

Screenshot 2020-07-01 at 12 41 16 PM

File: src/main/webapp/webapptests/spec/SpecRegistration.js [hunk 1]
```diff
@@ -65,69 +65,95 @@ describe("When displaying registration form", function() {
...
-        fetchLoginStatusHelper(document, mockLoginStatus);
+        // it("sets login link correctly", function() {
+        //     mockLoginStatus = {isLoggedIn:false, needsToRegister:false, url:'/_ah/login?continue=%2Fregistration.html'};
+        //     fetchLoginStatusHelper(document, mockLoginStatus);
 
-        expect(document.getElementById('login-form').style.display).toBe('block');
-        expect(document.getElementById('logout-form').style.display).toBe('none');
-        expect(document.getElementById('registration-form').style.display).toBe('none');
+        //     expect(document.getElementById('login-url').href).toBe('http://localhost:8080/_ah/login?continue=%2Fregistration.html');
+        // });
```
Comment (kashiish) [inline src/main/webapp/webapptests/spec/SpecRegistration.js:104]: Delete?

File: src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 2]
```diff
@@ -45,18 +47,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
-        setRegistered(response, name);
+        setRegistered(request, response, name);
     }
 
     /** Determines if a logged in user needs to register, created for testing */
-    public void setRegistered(HttpServletResponse response, String name) throws IOException {
+    public void setRegistered(HttpServletRequest request, HttpServletResponse response, String name) throws IOException {
         LoginStatus loginStatus;
+        // Gets the referring URL of the user so we can return them to the same page when they logout
+        String referrer = request.getHeader("referer");     //"referer" is intentionally misspelled
```
Comment (betrevisan) [inline src/main/java/com/google/sps/servlets/LoginStatusServlet.java:58]: I couldn't really find where header is being set. Could you explain really briefly how it worls?

Comment (samfalberg) [reply src/main/java/com/google/sps/servlets/LoginStatusServlet.java:58]: I'm not setting the header anywhere. getHeader("referer") just returns the address of whatever page the user was at before clicking the logout link.
For further information: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer

**Pull request merged**